Feature/profile

vyatsykiv opened the pull request.

vyatsykiv committed:
Profile screen:
- added changing status bar color on home pages
- added displaying user's avatar
- added flutter_svg library to display svg

vyatsykiv committed:
Merge remote-tracking branch 'origin/master' into feature/profile

# Conflicts:
#	lib/presentation/basket_route.dart
#	lib/presentation/location_route.dart
#	lib/presentation/orders_route.dart
#	lib/presentation/profile_route.dart
#	lib/presentation/shop_route.dart
#	lib/resources/app_theme.dart
#	pubspec.lock

vyatsykiv committed:
Merged master into feature/profile.

vyatsykiv committed:
Profile screen:
- added displaying user's info and experience
- added texts widgets

vyatsykiv committed:
Profile screen:
- added Achievements widget
- added Join the game button

vyatsykiv committed:
Profile screen:
- added Loyalty System and Setting cards

vyatsykiv committed:
Profile screen:
- added image for profile header.

vyatsykiv committed:
1. Removed unused font family.
2. Changed fonts on Profile screen.

ababicheva commented on lib/presentation/basket_route.dart:
```diff
@@ -1,9 +1,11 @@
 class BasketRoute extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
+    setWhiteStatusBar();
```
isn't it enough to set status bar color in main `CoffeeShop` widget?

vyatsykiv replied:
no, cause of statusBar color is different on different route (home page).

ababicheva commented on lib/resources/app_messages.dart:
```diff
@@ -9,4 +9,20 @@ class AppMessages {
   static const String ordersTitle = "Orders";
   static const String profileTitle = "Profile";
   static const String shopTitle = "Shop";
+  static const String stepsToFreeCoffee = "Steps to Free Coffee";
```
please add a small doc comment to visually separate strings.
e.g. `/// Profile Route`

ababicheva commented on lib/resources/app_text_styles.dart:
```diff
@@ -0,0 +1,4 @@
+class AppTextStyles {
```
this is not text styles, this is font family names.
text styles are: https://app.zeplin.io/project/5ebd52f3c3b62d4b25115ecd/screen/5ebe8e8738a1a0335abb2550
please rename this class.

ababicheva commented on lib/presentation/profile_route.dart:
```diff
@@ -1,9 +1,382 @@
+}
+
+class _UserAvatarWidget extends StatelessWidget {
+  final double userAvatarSize = 98;
```
let's use dimens for sizes

ababicheva commented on lib/resources/app_theme.dart:
```diff
@@ -23,4 +25,11 @@ const MaterialColor colorWhite = const MaterialColor(
\ No newline at end of file
+);
+
+void setWhiteStatusBar() {
```
let's make it a bit more generic by passing the color as an argument.
this will make it more understandable and we won't need to duplicate code in routes with specific status bar color like in `profile_route`
`SystemChrome.setSystemUIOverlayStyle(SystemUiOverlayStyle(
        statusBarColor: AppColors.colorBlackGray,
        statusBarIconBrightness: Brightness.light));`

yakivmospan commented on assets/images/bg_user_avatar.svg:
```diff
@@ -0,0 +1,3 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="97" height="97" viewBox="0 0 97 97">
```
Size of the images are directly set in svgs, can we override it somehow, for example by changing image widget size?

vyatsykiv replied:
Yes, SvgPicture.asset('imagePath', width, height)

yakivmospan commented on lib/widgets/text/uidisplay_widget.dart:
```diff
@@ -0,0 +1,18 @@
+import 'package:flutter/material.dart';
```
If we put commas to the end of the parameter list, then code formatting will look like this:

```
  UIDisplayWidget({
    String text,
    Color color,
    double fontSize,
    FontWeight fontWeight,
    double letterSpacing = 0.0,
  }) : super(
          text,
          style: TextStyle(
            fontFamily: AppTextStyles.SFUIDisplay,
            color: color,
            fontSize: fontSize,
            fontWeight: fontWeight,
            letterSpacing: letterSpacing,
          ),
        );
```

What would you guys prefer, keep it as is or update to the above? @vyatsykiv @budgieParrot @oleksandrakovalova @Igorkh106

vyatsykiv replied:
Probably yours version is better but in general it doesn't have any visual benefits for me.

yakivmospan replied:
We need to choose one and stay consistent

yakivmospan commented on lib/widgets/text/uitext_widget.dart:
```diff
@@ -0,0 +1,18 @@
+import 'package:flutter/material.dart';
+import 'package:flutterapp/resources/app_text_styles.dart';
+
+class UITextWidget extends Text {
```
What would you guys prefer for custom widget prefixes more, UITextWidget or `AppTextWidget`like we have with resources, styles etc.

@vyatsykiv @budgieParrot @oleksandrakovalova @Igorkh106

yakivmospan replied:
That also applies to DisplayWidget

vyatsykiv replied:
Name of font family is more logical for me.

yakivmospan replied:
But, we are passing a font inside... it can be anything not only UI family

yakivmospan replied:
Oops, sorry, we will pass `Style` inside.\

ababicheva replied:
it will become just AppTextWidget when I implement styles.
for me it's ok to be name like it is

yakivmospan commented on lib/widgets/achievement_widget.dart:
```diff
@@ -0,0 +1,76 @@
+      this.currentProgress,
+      this.progressColor,
+      this.icon,
+      this.progressBgColor = AppColors.colorShaleGray});
```
Please set a comma after the last parameter in the constructor, formatter will formant it better.

```
class AchievementWidget extends StatelessWidget {
  final String title;
  final String name;
  final int maxProgress;
  final int currentProgress;
  final Color progressColor;
  final Color progressBgColor;
  final String icon;

  AchievementWidget({
    this.title,
    this.name,
    this.maxProgress,
    this.currentProgress,
    this.progressColor,
    this.icon,
    this.progressBgColor = AppColors.colorShaleGray,
  });
```

ababicheva reviewed (commented):
Summary:

- comments in pull request
- move all dimens into separate {RouteName}Dimens file
- add new constructor to text widgets that will accept a textstyle

yakivmospan commented on lib/widgets/achievement_widget.dart:
```diff
@@ -0,0 +1,76 @@
+import 'package:flutterapp/widgets/text/uidisplay_widget.dart';
+import 'package:flutterapp/widgets/text/uitext_widget.dart';
+
+class AchievementWidget extends StatelessWidget {
```
Please create a separate package for profile route and put there:
- route
- all widgets that related only to that route and are in separate files, like an achievement

Lets keep only comment widgets in `widgets` package

yakivmospan reviewed (commented):
There are no selectors on:
- gift card
- bonuses
- payment methods
- sign out
- profile edit button

vyatsykiv committed:
Profile screen:
- added ExpandedAchievements widget

vyatsykiv committed:
1. Renamed AppTextStyles to AppFontFamilies.
2. Refactored setting StatusBar color.
3. Created 'profile' package and moved all related widget to it.

vyatsykiv committed:
Profile screen:
- added selectors for clickable items.

vyatsykiv committed:
Profile screen:
- moved all hardcoded dimens to ProfileDimens.dart and to AppDimens.dart.

vyatsykiv committed:
Added AppTextWidget.

vyatsykiv committed:
Profile screen:
- fixed displaying long user's name.

yakivmospan commented on lib/presentation/profile/profile_route.dart:
```diff
@@ -0,0 +1,682 @@
+  }
+}
+
+class _CardWidget extends StatelessWidget {
```
What if there will be more items on the card or first card will contain two and second three rows?

Maybe we can pass a list of rows to the Card? Something like:

`_CardItem(title, icon, listener)`

Where `_CardItem` can be a simple data class or a widget. And then pass a list of such items inside of `_CardWidget`. Will make it more universal.

```
_CardWidget(
    items: arrayOf(
        _CardItem(firstTitle, firstIcon, firstListener),
        _CardItem(secondTitle, secondIcon, secondListener),
    )
)
```

yakivmospan replied:
You can pass a list of `_CardRowWidget` that already exists

yakivmospan commented on lib/presentation/profile/profile_route.dart:
```diff
@@ -0,0 +1,682 @@
+    return _getProfileWidget();
+  }
+
+  Widget _getProfileWidget() {
```
Can you please investigate if this is still actual :https://iiro.dev/2018/12/11/splitting-widgets-to-methods-performance-antipattern/

yakivmospan replied:
Maybe this thread can have some more details: https://www.reddit.com/r/FlutterDev/comments/a5bug0/splitting_widgets_to_methods_is_a_performance/

yakivmospan commented on lib/presentation/profile/profile_route.dart:
```diff
@@ -0,0 +1,682 @@
+      color: AppColors.colorWhiteGray,
+      child: ListView(
+        children: [
+          Container(
```
Can you please move this container to a separate widget? Will be much easier to read main items that are on the screen then.

yakivmospan commented on lib/presentation/profile/profile_route.dart:
```diff
@@ -0,0 +1,682 @@
+    );
+  }
+
+  Widget _getExpandedWidget() {
```
Same as in the previous comment, can you please separate widgets a bit more, so there will be less inner wrappings in the one of main widgets on the screen?

vyatsykiv committed:
Profile screen:
- refactored expanding user's achievements.

vyatsykiv committed:
Profile screen:
- refactored creating cards.

vyatsykiv committed:
Merge remote-tracking branch 'origin/master' into feature/profile

# Conflicts:
#	lib/presentation/location_route.dart
#	lib/resources/app_colors.dart
#	lib/resources/app_text_letter_spacing.dart
#	lib/resources/app_text_sizes.dart
#	lib/resources/app_theme.dart

vyatsykiv committed:
Profile screen:
- merged TextStyles and new Colors.

vyatsykiv committed:
Profile screen:
- moved Card related widget to separate file.

vyatsykiv committed:
Profile screen:
- code polishing.

vyatsykiv committed:
Locked Portrait mode.

vyatsykiv committed:
Fixed issue with white status bar on Profile page on iOS.

vyatsykiv committed:
Created SimpleRippleWidget.

vyatsykiv committed:
1. Fixed displaying Profile page on small devices.
2. Fixed handling 'onTap' event in SimpleRippleWidget.

vyatsykiv committed:
Updated RippleWidget.
